### DISCUSSION TITLE
Support Terasology on ARM platform

### DISCUSSION THREAD

**Discussion by skaldarnar** [MovingBlocks/Terasology#4416] [Ideas]
Taking the discussion over from https://github.com/MovingBlocks/Terasology/issues/3412.

@DarkWeird @Adriankhl please feel free to provide more information on this 🤓

**Comment by DarkWeird** [comment]
We can provide arm support for linux-arm64 platform:

1. Current Lwjgl3 can provide it. (Just add natives to gradle.build)
2. Seems, bullet3 provides natives for arm64 too. (@pollend can you create JNBullet for arm?)
3. lua - don't known.

We cannot provide arm support for:
1. linux-arm32  - bullet not provides it
2. macos-arm - new platform, bullet3 and lwjgl3 not provides support (but macos-arm have layer for running x86 at arm)
3. windows-arm - not natives (should exists, don't sure)

**Reply by Adriankhl** [reply]
I think there shouldn't be any problem on lua. Lua (whichever version) is widely available at most arm linux distro.

**Comment by DarkWeird** [comment]
Pros:
1. linux-arm64 - runs on arm servers (should be low energy consume, profit) 
1.1. linux-arm64 - raspberry pi as client(?)
2. macos-arm - to support natively latest macoss should provide more performance, then translation layer for x86(_64)
3. windows-arm - i think windows-arm as desktop will appears soon (Nvidia arm, amd arm.. etc)
Cons:
1. more platforms for support

also we should to wait windows-arm and macos-arm jvm releases.
